### PR TITLE
fix interaction filters & add tests (DEV-1693)

### PR DESCRIPTION
**Pull request by mikefeldberg**
DEV-1693

old bug. caught while testing fe for release, [here](https://github.com/BetterAngelsLA/monorepo/pull/1203#discussion_r2033912346)

## Summary by Sourcery

Fix interaction filters for notes query to handle empty filter lists correctly

Bug Fixes:
- Modify filter logic to correctly handle empty lists for authors and teams filters in notes query, ensuring all notes are returned when no specific filter is applied

Tests:
- Add comprehensive test cases for notes query filters, including scenarios with empty author and team filter lists

### CODE REVIEW

File: apps/betterangels-backend/notes/types.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ class NoteFilter:
     def authors(
         self, queryset: QuerySet, info: Info, value: Optional[List[ID]], prefix: str
     ) -> Tuple[QuerySet[models.Note], Q]:
-        if value is None:
+        if value in [None, []]:
```
Comment (vecchp) [inline apps/betterangels-backend/notes/types.py:161]:
```python
t = None

if not t:
    print("t is None")

x = []

if not x:
  print('x is empty')
```
```
t is None
x is empty
```

I think we can simply do if not value here.

File: apps/betterangels-backend/notes/types.py [hunk 2]
```diff
@@ -191,7 +191,7 @@ def search(
     def teams(
         self, queryset: QuerySet, value: Optional[List[SelahTeamEnum]], prefix: str
     ) -> Tuple[QuerySet[models.Note], Q]:
-        if value is None:
+        if value in [None, []]:
```
Comment (vecchp) [inline apps/betterangels-backend/notes/types.py:194]: Same as above for here.

**Review by vecchp — CHANGES_REQUESTED** [review]

**Review by vecchp — APPROVED** [review]
